commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -18,34 +18,202 @@
+                        @Override
+                        public void onSuccess(@Nullable Configuration result) {
+                            if (result != null) {
+                                onGetConfigurationSuccess(context, result);
+                                Insight.getInsightTracker().trackImpression(context, pageName, pageGroup, params);
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(HyperwalletException exception) {
+                            // TODO unsure what to log here
+                            final String logMessage = MessageFormat.format(
+                                    "Unable to find Configuration using default token provider: {0}", exception);
+                        }
+
+                        @Override
+                        public Handler getHandler() {
+                            return null;
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    /**
+     * Used to track an intentional interaction by the user. Typically for interacting with a UI element such as Buttons
+     * and Spinners
+     *
+     * @param context   Context where the tracking is happening
+     * @param pageName  an arbituary pageName the user is currently viewing as defined by the app
+     * @param pageGroup an arbituary pageGroup the user is currently viewing as defined by the app
+     * @param link      the link the user is currently viewing as defined by the app
+     * @param params    additional information that can be added to make the tracking event more useful
+     */
+    public void trackClick(@NonNull final Context context, @NonNull final String pageName,
+            @NonNull final String pageGroup, @NonNull final String link, @NonNull final Map<String, String> params) {
+
+        if (Insight.getInsightTracker().isInitialized()) {
+            Insight.getInsightTracker().trackImpression(context, pageName, pageGroup, params);
+
+        } else {
+            mExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    Hyperwallet.getDefault().getConfiguration(new HyperwalletListener<Configuration>() {
+                        @Override
+                        public void onSuccess(@Nullable Configuration result) {
+                            if (result != null) {
+                                onGetConfigurationSuccess(context, result);
+                                Insight.getInsightTracker().trackImpression(context, pageName, pageGroup, params);
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(HyperwalletException exception) {
+                            // TODO unsure what to log here
+                            final String logMessage = MessageFormat.format(
```

Comment:
do nothing, we are not logging or something, if this Insight call is not successful, we will try to initialize it again in the next Insight invocation